src/app/utils/helper.ts

```diff
@@ -9,3 +9,12 @@ export const simpleDeepEqual = (firstObject: { [key: string]: any }, secondObjec
 
     return haveSameNumberOfKeys && haveSameValues;
 }
+
+export function getUniqueValuesOfKey(objects: { [key: string]: any }[], key: string): any[] {
+    const values = objects
+        .filter(o => key in o)
+        .reduce((prev: any[], curr) => prev.concat(curr[key]), [])
+        .filter(value => value != null);
+    const uniqueValues = new Set(values);
+    return [...uniqueValues];
```

Comment:
you should be able to do this in a single loop:

`return [...new Set(objects.map(item => item.key))];` and adjusting for missing keys, null values